client/src/SharedComponents/Slideshow/Slideshow.js

```diff
@@ -1,87 +1,94 @@
 import React, { Component } from "react";
-import { Slide } from "react-slideshow-image";
-import ProjectView from '../ProjectView'
 import StudentCard from '../StudentCard'
 import "./Slideshow.css";
+import Slider from "react-slick";
+import ProjectView from '../ProjectView'
 
 
-const dbProjects = ["","","","",""]
-const allProjects = dbProjects.map(project =>{
-return <ProjectView
-  projectImg={project}
-/>
-})
-
-const dbStudents = ["","","","",""]
-const allStudents = dbStudents.map(student =>{
-  return <StudentCard
-  studentImg= {student.img}
-   studentname={student.name}  />
-  })
-
-const properties = {
-  duration: 5000000000000000,
-  transitionDuration: 500,
-  infinite: false,
-  indicators: true,
-  arrows: true,
-};
-
 class Slideshow extends Component {
   state = {
-    projects: [],
-    students: []
+    projectSlide: [],
+    studentSlide: [],
   }
-  componentDidMount = () => {
-    let projectsSlide = [];
-    let studentsSlide = [];
-    for (let i = 0; i < allProjects.length; i += 3) {
-      projectsSlide.push(<div className="one-slide" key={Date.now()}>
-        {!allProjects[i]
-          ? <div></div>
-          : allProjects[i]
-        }
-        {!allProjects[i + 1]
-          ? <div></div>
-          : allProjects[i + 1]
-        }
-        {!allProjects[i + 2]
-          ? <div></div>
-          : allProjects[i + 2]
 
-        }
-      </div>)
-    }
-    for (let i = 0; i < allStudents.length; i += 3) {
-      studentsSlide.push(<div className="one-slide" key={Date.now()}>
-        <div className='student'>{allStudents[i]}</div>
-        <div className='student'>{allStudents[i + 1]}</div>
-        <div className='student'>{allStudents[i + 2]}</div>
-      </div>)
+  componentDidMount() {
+    const { projects, students } = this.props;
+
+    if (projects) {
+      const projectSlide = projects.map(project => {
+        return <ProjectView
+          projectImg={project.image}
+        />
+      })
+      this.setState({ projectSlide })
+
     }
+    if (students) {
```

Comment:
@Rahaf-96 You could use defaultProps for students and projects to guarantee that they are always an array:
```
Slideshow.defaultProps = {
  students: [],
  projects: []
}
```
and then you wouldn't need the `if (students) {` because map() will work correctly on an empty array.

Reply:
@tomduggan85  I changed it and it works ! thanks for the note :)

Reply:
You're welcome, glad it worked!